src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1668,16 +1668,16 @@ private static IEnumerable<CommandLineReference> ParseAssemblyReferences(string
                 alias = null;
             }
 
-            List<string> paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToList();
+            string[] paths = ParseSeparatedPaths(value).Where((path) => !string.IsNullOrWhiteSpace(path)).ToArray();
```

Comment:
Agree. In general ToList is better than ToArray from an allocation perspective. Is there a subtle issue we're missing here?